command/commands/team.py

```diff
@@ -8,97 +8,89 @@ class TeamCommand:
     """Represent Team Command Parser."""
 
     command_name = "team"
-    help = "# Team Command Reference\n\n"\
-           "/rocket team\n\n"\
-           "All Parameters with whitespace"\
-           " must be enclosed by quotation marks.\n\n"\
-           "## Options to specify input\n\n"\
-           "* list\n"\
-           "    * outputs the Github team names"\
-           " and display names of all teams\n\n"\
-           "* view GITHUB_TEAM_NAME\n"\
-           "    * view information and members of a team\n\n"\
-           "* help\n"\
-           "    * outputs options for team commands\n\n"\
-           "## TEAM LEAD or ADMIN only\n\n"\
-           "* create GITHUB_TEAM_NAME"\
-           " [--name DISPLAY_NAME]"\
-           " [--platform PLATFORM]"\
-           " [--channel]\n"\
-           "    * create a new team with a Github team name"\
-           " and optional parameters\n"\
-           "    * the user will be automatically added to the new team\n"\
-           "    * if --channel flag included, add all members in channel\n\n"\
-           "The following can only be used by"\
-           " a team lead in the team or an admin:\n\n"\
-           "* edit GITHUB_TEAM_NAME"\
-           " [--name DISPLAY_NAME]"\
-           " [--platform PLATFORM]\n"\
-           "    * edit properties of specified team\n\n"\
-           "* add GITHUB_TEAM_NAME SLACK_ID\n"\
-           "    * add the specified user to the team\n\n"\
-           "* remove GITHUB_TEAM_NAME SLACK_ID\n"\
-           "    * remove the specified user from the team\n\n"\
-           "* delete GITHUB_TEAM_NAME\n"\
-           "    * permanently delete the specified team\n"
     desc = "for dealing with " + command_name + "s"
 
     def __init__(self):
         """Initialize team command parser."""
         logging.info("Initializing TeamCommand instance")
         self.desc = ""
-        self.parser = argparse.ArgumentParser(prog="team")
+        self.parser = argparse.ArgumentParser(prog="/rocket")
         self.parser.add_argument("team")
-        self.init_subparsers()
+        self.subparser = self.init_subparsers()
+        self.help = self.get_help()
 
     def init_subparsers(self):
         """Initialize subparsers for team command."""
         subparsers = self.parser.add_subparsers(dest="which")
 
         """Parser for list command."""
         parser_list = subparsers.add_parser("list")
-        parser_list.set_defaults(which="list")
+        parser_list.set_defaults(which="list",
+                                 help="outputs the Github team names "
```

Comment:
Not a change to be made, but I wonder if there's a way to get these help strings programmatically - perhaps there's a way to read the docstrings directly?

Reply:
there is a way yep, actually i think that's what my get_help is doing, just iterating over each command and appending their individual help strings

Reply:
Sorry, I mean, it looks like we have to set this `help` parameter by hand. I'm wondering if there's a way to set it in an automated fashion, perhaps by reading the docstrings we write for each handler directly. This would probably be quite a big task though, so I wouldn't bother with it for now.

Reply:
o ok i see